Replace relative links

bsquizz opened the pull request:
Replaces relative links found in markdown files using the document set's citation URL as a base URL. A relative link, e.g.:

```
[Link](../something.md)
```

will be transformed into an absolute link, e.g.:
```
[Link](https://citation.url/path/to/docs/../something.md)
```

adamrdrew reviewed (approved):
This LGTM. The only concern I have is if there were malformed links, like an empty one, or a link without a target, or even just some text that could be misinterpreted as a link would it bomb out? I think it would be OK but maybe having like a try/catch or something around the call to `_convert_relative_links` - but I might be over thinking it

bsquizz commented:
In theory the regex should not match unless there's actual content in the `[]` or the `()`. But who knows, it is true that there might be a corner case missed. I added a unit test to cover the "bad links" scenarios I can picture, but I also added a try/except to be safe too.
